pkg/services/gitopswriter/gitopswriter.go

```diff
@@ -0,0 +1,211 @@
+package gitopswriter
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/fluxcd/go-git-providers/gitprovider"
+	"github.com/weaveworks/weave-gitops/pkg/git"
+	"github.com/weaveworks/weave-gitops/pkg/gitproviders"
+	"github.com/weaveworks/weave-gitops/pkg/logger"
+	"github.com/weaveworks/weave-gitops/pkg/models"
+	"github.com/weaveworks/weave-gitops/pkg/osys"
+	"github.com/weaveworks/weave-gitops/pkg/services/automation"
+	"github.com/weaveworks/weave-gitops/pkg/services/gitrepo"
+
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	AddCommitMessage    = "Add application manifests"
+	RemoveCommitMessage = "Remove application manifests"
+)
+
+var _ GitOpsDirectoryWriter = &GitOpsDirectoryWriterSvc{}
+
+type GitOpsDirectoryWriter interface {
+	AddApplication(ctx context.Context, app models.Application, clusterName string, autoMerge bool) error
+	RemoveApplication(ctx context.Context, app models.Application, clusterName string, autoMerge bool) error
+}
+
+type GitOpsDirectoryWriterSvc struct {
+	Automation automation.AutomationGenerator
+	RepoWriter gitrepo.RepoWriter
+	Osys       osys.Osys
+	Logger     logger.Logger
+}
+
+func NewGitOpsDirectoryWriter(automationSvc automation.AutomationGenerator, repoWriter gitrepo.RepoWriter, osys osys.Osys, logger logger.Logger) GitOpsDirectoryWriter {
+	return &GitOpsDirectoryWriterSvc{Automation: automationSvc, RepoWriter: repoWriter, Osys: osys, Logger: logger}
+}
+
+func (dw *GitOpsDirectoryWriterSvc) AddApplication(ctx context.Context, app models.Application, clusterName string, autoMerge bool) error {
+	manifests, err := dw.Automation.GenerateAutomation(ctx, app, clusterName)
+	if err != nil {
+		return fmt.Errorf("could not generate application GitOps Automation manifests: %w", err)
+	}
+
+	remover, repoDir, err := dw.RepoWriter.CloneRepo(ctx, app.Branch)
+	if err != nil {
+		return fmt.Errorf("failed to clone repo: %w", err)
+	}
+
+	defer remover()
+
+	kManifest, err := addKustomizeResources(app, repoDir, clusterName, appKustomizeReference(app))
+	if err != nil {
+		return err
+	}
+
+	manifests = append(manifests, kManifest)
+
+	if !autoMerge {
```

Comment:
minor: invert the logic to make easier to read

```go
if autoMerge {
```

Reply:
:+1: